Update package versions in conda recipe

xylar opened the pull request:
A new build with these packages will help with an issue related to weird package version names for `esmf` that have been giving me trouble with `conda>=4.6` and `cdms2=3.0.1`.

xylar commented:
@zshaheen, I was able to build on my laptop with these versions and `conda 4.6.4`.  Please let me know if they work for you (with whatever `conda` you have).  If so, please upload this new build to the `e3sm` channel. As you may know, the current build (`0`) will still be there and will be used if need be (e.g. if package compatibility requires it) so there should be no harm in uploading a second build of `1.6.0`.

xylar commented:
Obviously, it would be good to test this build before uploading it if that's practical.

xylar commented:
Unrelated question but I'm wondering if you know why the repo is 1.2 GB in size (and if there's anything one could do to clean it up).  It takes ages to download and ages more to build packages from it.  Did you store some really big data sets in it at one point and they're still in the history?

xylar commented:
Hmm, not as simple as I was hoping.  I'll look into the linking error exposed by Travis-CI.

xylar commented:
@zshaheen and @chengzhuzhang, am I right in thinking that your travis-ci hasn't been working in awhile and maybe I shouldn't work to hard to get it working here?  I see red x's on all your recent PR merges.  Might be worth getting it working again but that's beyond the scope of this PR.  Are there other tests I (or one of you) can run to see if this new set of packages works as expected?

zshaheen commented:
@xylar, thanks for doing this. I'm going to test the changes you've made to the `meta.yaml` and hopefully get a working env. Also, don't worry about any CI, getting it running isn't a real priority now.

xylar commented:
@zshaheen  and @chengzhuzhang, this recipe should help make `e3sm_diags` work in the latest `e3sm-unified` that I'm working on.  I've done several things that I'd like you to take a look at and let me know if this fits with your plans and design philosophy:
* I changed the recipe to "noarch" (I'll point this out in a comment below) so that a single build can be used on linux and mac (and possibly windows) and also python 2.7, 3.6 and 3.7.  Saves a lot of work for packages that don't need separate builds, which I think is true of `e3sm_diags`
* Changed to the recommended [build command](http://conda-forge.org/docs/maintainer/adding_pkgs.html#build) from conda-forge
* changed the dependency pinnings from `==` to `>=` to accommodate future updates like just happened for `cdat` without requiring a new build.

Please check if the versions are sensible.  Please build the package and test to make sure it, in fact, works and upload it to the `e3sm` channel at our earliest convenience so I can build the next `e3sm-unified` soon.

xylar commented on .travis.yml:
```diff
@@ -15,15 +15,18 @@ before_install:
   - bash miniconda.sh -b -p $HOME/miniconda
-  - export PATH="$HOME/miniconda/bin:$PATH"
+  - source $HOME/miniconda/etc/profile.d/conda.sh
+  - conda activate base
```
This is by far the preferred way of activating a conda environment, not setting the `PATH`.  Certain environment variables don't get set if `conda activate` is not called.

xylar commented on .travis.yml:
```diff
@@ -15,15 +15,18 @@ before_install:
 
-  - conda install python=$TRAVIS_PYTHON_VERSION cdp=1.1.0 vcs=2.12 vtk-cdat=7.1.0.2.12 vcsaddons=2.12 dv3d=2.12 cdms2=2.12 cdutil=2.12 genutil=2.12 cdtime=2.12 numpy "matplotlib<=2.0.2" cartopy beautifulsoup4 lxml flake8 -c conda-forge -c uvcdat -y
+  - conda create -n e3sm_diags python=$TRAVIS_PYTHON_VERSION cdp=1.4.2 vcs=8.1 vtk-cdat=8.2.0rc2.8.1 vcsaddons=8.1 dv3d=8.1 cdms2=3.1.2 cdutil=8.1 genutil=8.1.1 cdtime=3.1.2 numpy matplotlib cartopy beautifulsoup4 lxml flake8 -c conda-forge -c cdat -c cdat/label/v81 -y
+  - conda activate e3sm_diags
```
Much better to create a new environment than to install into the base environment.

xylar commented on conda/conda_build_config.yaml:
```diff
@@ -1,3 +1,4 @@
 python:
     - 2.7
     - 3.6
+    - 3.7
```
I assume you want to be able to support python 3.7, too, right?

xylar commented on conda/meta.yaml:
```diff
@@ -7,24 +7,27 @@ source:
 
 build:
-    script: python setup.py install
+    number: 1
```
Add the build number so we can increase it to 1 (since you already have a build of 1.6.0 on the `e3sm` channel).

xylar commented on conda/meta.yaml:
```diff
@@ -7,24 +7,27 @@ source:
 build:
-    script: python setup.py install
+    number: 1
+    noarch: python
```
By adding this line, you don't need separate builds/uploads to anaconda for different platforms (linux, mac and windows) or different python versions (2.7, 3.6 and 3.7).

xylar commented on conda/meta.yaml:
```diff
@@ -7,24 +7,27 @@ source:
-    script: python setup.py install
+    number: 1
+    noarch: python
+    script: "{{ PYTHON }} -m pip install . --no-deps -vv"
```
This is the preferred build command (see http://conda-forge.org/docs/maintainer/adding_pkgs.html#build).

xylar commented on conda/meta.yaml:
```diff
@@ -7,24 +7,27 @@ source:
+    - python
     - setuptools
+    - pip
 
```
These belong in `host` rather than `build` according to conda-forge documentation.  `pip` is required for the preferred build command above.

xylar commented on conda/meta.yaml:
```diff
@@ -7,24 +7,27 @@ source:
+    - cdms2 >=3.1.2
+    - cdutil >=8.0
+    - genutil >=8.1
+    - cdtime >=3.1.2
```
Many versions were changed.  Because `vtk-cdat` is a release candidate, I have to pin it explicitly.  All other versions, I changed to >= where I was able.

xylar commented on conda/meta.yaml:
```diff
@@ -7,24 +7,27 @@ source:
+    - cdms2 >=3.0.1
+    - cdutil >=8.0
+    - genutil >=8.0
+    - cdtime >=3.0
```
Because vtk-cdat is a release candidate, I have to pin it explicitly. All other versions, I changed to `>=` to allow the latest `cdat` to work.

xylar commented:
It would be a good idea to add some tests to your conda recipe (even if they are as simple as making sure the conda package can be imported or the `--help` option on your entry point or scripts works as expected).

zshaheen commented:
@xylar, thanks for doing this! So I successfully built it myself, but I'm getting old versions of some software.

@doutriaux1, I'm installing via the command below:

```
conda create -n e3sm_diags_env_1.6.0_build1 -c zshaheen -c conda-forge -c cdat e3sm_diags=1.6.0
```

I'm getting an older version of `cdms2`, which is causing issues.
```
(e3sm_diags_env_1.6.0_build1) shaheen2@acme1:~/github/acme_diags$ conda list cdms
# packages in environment at /export/shaheen2/anaconda2/envs/e3sm_diags_env_1.6.0_build1:
#
# Name                    Version                   Build  Channel
cdms2                     3.0.0                    pypi_0    pypi
libcdms                   3.1.2             hcbdc9ef_1000    conda-forge
```

Any input Charles? It's weird that we're getting something from the `pypi` channel. It might be due to some of the changes that @xylar made, but I'm not too sure.

zshaheen commented:
In the `meta.yaml` for this PR, we specifically have a greater version of `cdms2`.

```
  run:
    - cdms2 >=3.0.1
```

xylar commented:
@zshaheen, you need to use the `-c cdat/label/v81` channel in addition to the `-c cdat` channel.

doutriaux1 commented:
don't worry about 3.0.0 it's "Wrong" due to cdms setup.py not picking up the git tag. But it actually brings the correct version (Lina, Denis and I double checked)

C.


From: Zeshawn Shaheen <notifications@github.com>
Reply-To: E3SM-Project/e3sm_diags <reply@reply.github.com>
Date: Monday, March 4, 2019 at 12:51 PM
To: E3SM-Project/e3sm_diags <e3sm_diags@noreply.github.com>
Cc: "Doutriaux, Charles" <doutriaux1@llnl.gov>, Mention <mention@noreply.github.com>
Subject: Re: [E3SM-Project/e3sm_diags] Update package versions in conda recipe (#203)


@xylar<https://github.com/xylar>, thanks for doing this! So I successfully built it myself, but I'm getting old versions of some software.

@doutriaux1<https://github.com/doutriaux1>, I'm installing via the command below:

conda create -n e3sm_diags_env_1.6.0_build1 -c zshaheen -c conda-forge -c cdat e3sm_diags=1.6.0

I'm getting an older version of cdms2, which is causing issues.

(e3sm_diags_env_1.6.0_build1) shaheen2@acme1:~/github/acme_diags$ conda list cdms

# packages in environment at /export/shaheen2/anaconda2/envs/e3sm_diags_env_1.6.0_build1:

#

# Name                    Version                   Build  Channel

cdms2                     3.0.0                    pypi_0    pypi

libcdms                   3.1.2             hcbdc9ef_1000    conda-forge

Any input Charles? It's weird that we're getting something from the pypi channel. It might be due to some of the changes that @xylar<https://github.com/xylar> made, but I'm not too sure.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/E3SM-Project/e3sm_diags/pull/203#issuecomment-469416689>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ACpw8eRUOmqSwOLKkJgGOYe2UxH3Wo0Gks5vTYdKgaJpZM4bEc7k>.

xylar commented:
> you need to use the -c cdat/label/v81 channel in addition to the -c cdat channel.

I don't remember what still comes from the `cdat` channel but a few of the packages that are not dependencies of the main `cdat` package do.  That may be true for `e3sm-unified` but for `e3sm_diags`, you might not need the main `cdat` channel anymore.

xylar commented:
> don't worry about 3.0.0 it's "Wrong" due to cdms setup.py not picking up the git tag. But it actually brings the correct version (Lina, Denis and I double checked)

I disagree.  If you're picking up the version from pypi, something's definitely wrong.

doutriaux1 commented:
it's not from pypi (it's not avail there) it's an artefact of cdms install

From: Xylar Asay-Davis <notifications@github.com>
Reply-To: E3SM-Project/e3sm_diags <reply@reply.github.com>
Date: Monday, March 4, 2019 at 12:58 PM
To: E3SM-Project/e3sm_diags <e3sm_diags@noreply.github.com>
Cc: "Doutriaux, Charles" <doutriaux1@llnl.gov>, Mention <mention@noreply.github.com>
Subject: Re: [E3SM-Project/e3sm_diags] Update package versions in conda recipe (#203)


don't worry about 3.0.0 it's "Wrong" due to cdms setup.py not picking up the git tag. But it actually brings the correct version (Lina, Denis and I double checked)

I disagree. If you're picking up the version from pypi, something's definitely wrong.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/E3SM-Project/e3sm_diags/pull/203#issuecomment-469418821>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ACpw8a-kX7kwigOPMEg1-lDQC_10ULY3ks5vTYjSgaJpZM4bEc7k>.

zshaheen commented:
So I did: `conda create -n e3sm_diags_env_1.6.0_build1_try2 -c zshaheen -c conda-forge -c cdat/label/v81 -c cdat e3sm_diags=1.6.0` with no luck.

```
shaheen2@acme1:~$ conda activate e3sm_diags_env_1.6.0_build1_try2
(e3sm_diags_env_1.6.0_build1_try2) shaheen2@acme1:~$ conda list cdms
# packages in environment at /export/shaheen2/anaconda2/envs/e3sm_diags_env_1.6.0_build1_try2:
#
# Name                    Version                   Build  Channel
cdms2                     3.0.0                    pypi_0    pypi
libcdms                   3.1.2             hcbdc9ef_1000    conda-forge
```

This old version of cdms is giving me errors.

doutriaux1 commented:
No matter what channel you use it WILL say 3.0.0 it is a bug, but it is NOT 3.0.0 do not worry about it.

C.


From: Zeshawn Shaheen <notifications@github.com>
Reply-To: E3SM-Project/e3sm_diags <reply@reply.github.com>
Date: Monday, March 4, 2019 at 1:35 PM
To: E3SM-Project/e3sm_diags <e3sm_diags@noreply.github.com>
Cc: "Doutriaux, Charles" <doutriaux1@llnl.gov>, Mention <mention@noreply.github.com>
Subject: Re: [E3SM-Project/e3sm_diags] Update package versions in conda recipe (#203)


So I did: conda create -n e3sm_diags_env_1.6.0_build1_try2 -c zshaheen -c conda-forge -c cdat/label/v81 -c cdat e3sm_diags=1.6.0 with no luck.

shaheen2@acme1:~$ conda activate e3sm_diags_env_1.6.0_build1_try2

(e3sm_diags_env_1.6.0_build1_try2) shaheen2@acme1:~$ conda list cdms

# packages in environment at /export/shaheen2/anaconda2/envs/e3sm_diags_env_1.6.0_build1_try2:

#

# Name                    Version                   Build  Channel

cdms2                     3.0.0                    pypi_0    pypi

libcdms                   3.1.2             hcbdc9ef_1000    conda-forge

This old version of cdms is giving me errors.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/E3SM-Project/e3sm_diags/pull/203#issuecomment-469430921>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ACpw8Z1YYcM2lIoeKQQ6taZoye0vjLgHks5vTZGhgaJpZM4bEc7k>.

doutriaux1 commented:
Zeshawn,

When you type this line it should be showing you a list of the pkg it's going to download and from where (before you press "y")

What does it say for cdms? You might need to add: --show-channel-urls

C.


From: Zeshawn Shaheen <notifications@github.com>
Reply-To: E3SM-Project/e3sm_diags <reply@reply.github.com>
Date: Monday, March 4, 2019 at 1:35 PM
To: E3SM-Project/e3sm_diags <e3sm_diags@noreply.github.com>
Cc: "Doutriaux, Charles" <doutriaux1@llnl.gov>, Mention <mention@noreply.github.com>
Subject: Re: [E3SM-Project/e3sm_diags] Update package versions in conda recipe (#203)

conda create -n e3sm_diags_env_1.6.0_build1_try2 -c zshaheen -c conda-forge -c cdat/label/v81 -c cdat e3sm_diags=1.6.0

zshaheen commented:
How strange, you're correct.

```
shaheen2@acme1:~$ conda create -n e3sm_diags_env_1.6.0_build1_try2 -c zshaheen -c conda-forge -c cdat/label/v81 -c cdat e3sm_diags=1.6.0
Collecting package metadata: done
Solving environment: done

## Package Plan ##

  environment location: /export/shaheen2/anaconda2/envs/e3sm_diags_env_1.6.0_build1_try2

  added / updated specs:
    - e3sm_diags=1.6.0


The following NEW packages will be INSTALLED:

  ...
  cdms2              conda-forge/linux-64::cdms2-3.1.2-py36h6091dcd_7
  ...
```

xylar commented:
Thanks for the clarification, @doutriaux1.  It's maybe worth doing build 1 of the package to fix this issue when you have time, since it's obviously a point of confusion.

doutriaux1 commented:
I told you, it's the correct cdms, but conda env will report the 3.0.0 in the yaml .It's a "feature" I showed Denis where it comes from last week. I'm sure it will be fixed eventually. But It's not a high priority.

C>


From: Zeshawn Shaheen <notifications@github.com>
Reply-To: E3SM-Project/e3sm_diags <reply@reply.github.com>
Date: Monday, March 4, 2019 at 2:00 PM
To: E3SM-Project/e3sm_diags <e3sm_diags@noreply.github.com>
Cc: "Doutriaux, Charles" <doutriaux1@llnl.gov>, Mention <mention@noreply.github.com>
Subject: Re: [E3SM-Project/e3sm_diags] Update package versions in conda recipe (#203)


How strange, you're correct.

shaheen2@acme1:~$ conda create -n e3sm_diags_env_1.6.0_build1_try2 -c zshaheen -c conda-forge -c cdat/label/v81 -c cdat e3sm_diags=1.6.0

Collecting package metadata: done

Solving environment: done



## Package Plan ##



  environment location: /export/shaheen2/anaconda2/envs/e3sm_diags_env_1.6.0_build1_try2



  added / updated specs:

    - e3sm_diags=1.6.0





The following NEW packages will be INSTALLED:



  ...

  cdms2              conda-forge/linux-64::cdms2-3.1.2-py36h6091dcd_7

  ...

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/E3SM-Project/e3sm_diags/pull/203#issuecomment-469438773>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ACpw8erBWLYffF7SJeCqmwq4NqwAG5L6ks5vTZdhgaJpZM4bEc7k>.

chengzhuzhang commented:
I used the py3_support branch for the testing. I tested the envs generated by:
```
conda create -n e3sm_diags_env_1.6.0_build1_try2 -c zshaheen -c conda-forge -c cdat/label/v81 -c cdat e3sm_diags=1.6.0
```
on both Mac and Linux. 
For Mac, it picks up python 2; For Linux, it picks up python 3.

The system tests failed in both env, with similar error:
```
" from . import _cdtime
ImportError: libcdms.so: cannot open shared object file: No such file or directory"
```

and
```
   from . import _cdtime
ImportError: dlopen(/Users/zhang40/anaconda2/envs/e3sm_diags_env_1.6.0_build1_try2/lib/python2.7/site-packages/cdtime/_cdtime.so, 2): Library not loaded: @rpath/libcdms.dylib
  Referenced from: /Users/zhang40/anaconda2/envs/e3sm_diags_env_1.6.0_build1_try2/lib/python2.7/site-packages/cdtime/_cdtime.so
  Reason: image not found
```
The problem is old cdutil (8.0) was picked up. I had this problem before with 8.0. I think what needed to be figured out are why 8.0 but not 8.1 are choosed by conda...

xylar commented:
@chengzhuzhang, I tried exactly the command you posed above on linux.  I get python 3.6 and cdutil 8.1.  I am starting with a python 3.6 version of miniconda.  Could it be that yours is python 3.7 and that makes a difference?

xylar commented:
I think it would be the best practice to always specify (and ask your users to specify) the python version when creating a conda environment.  Otherwise, you don't know for sure what you'll get.  But we need `e3sm_diags` to work for python 2.7, 3.6 and 3.7 all the same way.

xylar commented:
Here's the error I get with my `e3sm_diags` environment created with @chengzhuzhang's command above:
```
$ e3sm_diags --help
Traceback (most recent call last):
  File "/home/xylar/miniconda3/envs/e3sm_diags_env_1.6.0_build1_try2/bin/e3sm_diags", line 6, in <module>
    from acme_diags.acme_diags_driver import main
  File "/home/xylar/miniconda3/envs/e3sm_diags_env_1.6.0_build1_try2/lib/python3.6/site-packages/acme_diags/acme_diags_driver.py", line 24, in <module>
    from acme_diags.acme_viewer import create_viewer
  File "/home/xylar/miniconda3/envs/e3sm_diags_env_1.6.0_build1_try2/lib/python3.6/site-packages/acme_diags/acme_viewer.py", line 16, in <module>
    from acme_diags.driver.utils.general import get_set_name
  File "/home/xylar/miniconda3/envs/e3sm_diags_env_1.6.0_build1_try2/lib/python3.6/site-packages/acme_diags/driver/utils/__init__.py", line 1, in <module>
    import general
ModuleNotFoundError: No module named 'general'
```

zshaheen commented:
> Here's the error I get with my `e3sm_diags` environment created with @chengzhuzhang's command above:
> 
> ```
> $ e3sm_diags --help
> Traceback (most recent call last):
>   File "/home/xylar/miniconda3/envs/e3sm_diags_env_1.6.0_build1_try2/bin/e3sm_diags", line 6, in <module>
>     from acme_diags.acme_diags_driver import main
>   File "/home/xylar/miniconda3/envs/e3sm_diags_env_1.6.0_build1_try2/lib/python3.6/site-packages/acme_diags/acme_diags_driver.py", line 24, in <module>
>     from acme_diags.acme_viewer import create_viewer
>   File "/home/xylar/miniconda3/envs/e3sm_diags_env_1.6.0_build1_try2/lib/python3.6/site-packages/acme_diags/acme_viewer.py", line 16, in <module>
>     from acme_diags.driver.utils.general import get_set_name
>   File "/home/xylar/miniconda3/envs/e3sm_diags_env_1.6.0_build1_try2/lib/python3.6/site-packages/acme_diags/driver/utils/__init__.py", line 1, in <module>
>     import general
> ModuleNotFoundError: No module named 'general'
> ```

I fixed this issue on the [`py3_support`](https://github.com/E3SM-Project/e3sm_diags/tree/py3_support) branch. There are others however, with some issues regarding `cdms2`. We're working on this.

xylar commented:
Okay so I shouldn't expect 1.6.0 to work with python 3?

zshaheen commented:
@xylar No, we'll release a new version after the mentioned issue is resolved, which is hopefully soon.

xylar commented:
Okay, I see.  Keep me posted.

xylar commented:
@zshaheen and @chengzhuzhang, I'm wondering what the status is.  I've been hoping to have a new release of e3sm-unified for awhile now and since cdat got updated, e3sm_diags (and processflow through its dependency on e3sm_diags) is the only package I'm aware of that isn't ready.  I realize you're both preparing for the meeting next week but I will only have 3 day after returning from Denver before I am out for 2 weeks of travel and won't be able to work on e3sm-unified.  That means basically that a release happens in the next week or so, or not for about another month.

Is there an issue beyond updating dependencies that is preventing e3sm_diags from working with cdat and co. v8.1?

chengzhuzhang commented:
Hi @xylar, Zeshawn reported last Friday that the conda recipe should be ready. It took some time to make it Python 3 compatible again since some dependencies were updated. @zshaheen  would you please double check the status. Please let Xylar know if everything is ready.

xylar commented:
@chengzhuzhang and @zshaheen, that sounds perfect to me!  Sorry for adding stress.

zshaheen commented:
Hi @xylar, apologies for the delay. My initial plan was to get both the Python 2 and Python 3 versions of the software working, but that's not possible. More time is needed for the Python 3 version.

I'm currently testing the Python 2 version. Since the `python setup.py install` line was removed from the `meta.yaml`, the correct files are no longer being moved to `/path/to/anaconda2/envs/the_env/share/`. Since `{{ PYTHON }} -m pip install . --no-deps -vv` is the recommended way, is there a way go get our functionality that we did in `setup.py` to work?

xylar commented:
@zshaheen, oh, I missed that you had some non-setuptools parts to your setup.py.   Let me take a closer look tonight but I think there is a way to do what you're trying to do in `setup.py` in a more "correct" way.  I assumed that `pip install` would be calling `setup.py` but maybe your custom steps aren't doing what you expect. I'll get back to you.

xylar commented:
I think the right way to do this is with `data_files`, see:
https://docs.python.org/3.3/distutils/setupscript.html#installing-additional-files

I'll see if I can make the appropriate modification tonight to show you what you want to be doing.

xylar commented:
Oops, sorry, that's what you're already doing.

xylar commented on setup.py:
```diff
@@ -33,7 +33,7 @@ def get_all_files_in_dir(directory, pattern):
 rgb_files = get_all_files_in_dir('acme_diags/plot/colormaps', '*.rgb')
 control_runs_files = get_all_files_in_dir('acme_diags/driver/control_runs', '*.csv')
 
-INSTALL_PATH = os.path.join(sys.prefix, 'share/e3sm_diags/')
```
@zshaheen, you can't use the current `sys.prefix` as the install path because the package is built in one environment (and usually on a completely different machine) than the one where it will be installed.  So if I build the package in my base environment, the scripts also get installed in `shared/e3sm_diags` in my base environment, which is not what I want and the package is broken for everyone else because they don't have a `/home/xylar/miniconda3/` path.  The solution is just to leave it as a relative path, and it will automatically have the install prefix prepended.

xylar commented:
@zshaheen, see my comment above.  Because `setup.py` needs to be modified, this build script will no longer work with the `v1.6.0` and you'll need to do a `v1.6.1` release to get a working package.

By the way, I had no trouble installing on linux with both python 2.7 and 3.7.  Are the issues with python 3 related to running the code and not just getting an environment?

zshaheen commented:
@xylar, I'll create a temp release on my conda channel and build a conda package with your new commits and test it.

> By the way, I had no trouble installing on linux with both python 2.7 and 3.7. Are the issues with python 3 related to running the code and not just getting an environment?

It was an issue when running the software. That's pretty much fixed on the `py3_support` branch. Since we're probably going to use this Python 3 env as the default from this point on, I want to make sure that our dependencies are working solidly before continuing.

xylar commented:
> It was an issue when running the software. That's pretty much fixed on the py3_support branch. Since we're probably going to use this Python 3 env as the default from this point on, I want to make sure that our dependencies are working solidly before continuing.

Okay, great!  Take your time and let me know when you're confident.

zshaheen commented:
@xylar and @chengzhuzhang. There seems to be some confusion as to what's happening. Below is the progress:

* Python 3: Our code is compatible on the `py3_support` branch. We just need to test the dependencies, since this is the first time many of them will have their Python 3 version used extensively.
  * So this will be supported later, after you do another release of the unified env Xylar.
* Python 2: Obviously, our code works with Python 2, it's what we've been testing/using since the beginning. I've tested your new changes and they work.
  * **However, when testing e3sm_diags with vcs 8.0 and vtk-cdat 8.2.0rc2.8.1 doesn't work.**

---

@doutriaux1 says that if we use vcs 8.1, it'll work with the version of vtk-cdat you've posted. I'm awaiting his input via email on this.

zshaheen commented:
@xylar, will you have v8.1 of vcs in the unified env? If so, I can merge and release v1.6.1 of e3sm_diags. You can then include this new version in the unified.

xylar commented:
> will you have v8.1 of vcs in the unified env? If so, I can merge and release v1.6.1 of e3sm_diags. You can then include this new version in the unified.

Yes, I'll use that version.  Here's the current plan for the pinned versions:
https://github.com/E3SM-Project/e3sm-unified/pull/39/files#diff-41173e91856f14b891cbdf2349423a83R34

zshaheen commented:
@xylar, Sounds good! I'm merging this PR and will ping you when we have a new release of e3sm_diags.

In a few weeks, we'll let you know when we have a working Python 3 version.
